packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/creditCard.gql.js

```diff
@@ -161,3 +161,10 @@ export default {
         setCreditCardDetailsOnCartMutation: SET_CC_DETAILS_ON_CART
     }
 };
+export const creditCardResolvers = {
+    Cart: {
+        paymentNonce: cart => {
+            return cart.paymentNonce || null;
```

Comment:
Whenever you write a `@client` field you have to define a resolver for it or Apollo "breaks" without an error. In my experience I found that this would result in a successful network call for the non-client fields but the `data` response would always be undefined after the query/mutation was completed.

Reply:
Good to know

Reply:
To come back to this, I was sort of wrong here - You either need to have defined a local resolver or prime your cache with a default/initial state. Since we don't write a cart object to cache on initialization we must define local resolvers and have a try-catch since the `readQuery` will throw an error the first time:

https://www.apollographql.com/docs/react/data/local-state/#querying-local-state